src/legacy/core_plugins/kibana/public/dashboard/panel/panel_utils.ts

```diff
@@ -82,38 +109,40 @@ export class PanelUtils {
     return panel;
   }
 
-  static parseVersion(version = '6.0.0') {
+  public static parseVersion(version = '6.0.0') {
     const versionSplit = version.split('.');
     if (versionSplit.length < 3) {
-      throw new Error(i18n.translate('kbn.dashboard.panel.invalidVersionErrorMessage', {
-        defaultMessage: 'Invalid version, {version}, expected {semver}',
-        values: {
-          version,
-          semver: '<major>.<minor>.<patch>',
-        },
-      }));
+      throw new Error(
+        i18n.translate('kbn.dashboard.panel.invalidVersionErrorMessage', {
+          defaultMessage: 'Invalid version, {version}, expected {semver}',
+          values: {
+            version,
+            semver: '<major>.<minor>.<patch>',
+          },
+        })
+      );
     }
     return {
       major: parseInt(versionSplit[0], 10),
-      minor: parseInt(versionSplit[1], 10)
+      minor: parseInt(versionSplit[1], 10),
     };
   }
 
-  static initPanelIndexes(panels) {
+  public static initPanelIndexes(panels: PanelState[]) {
     // find the largest panelIndex in all the panels
     let maxIndex = this.getMaxPanelIndex(panels);
 
     // ensure that all panels have a panelIndex
-    panels.forEach(function (panel) {
+    panels.forEach(panel => {
       if (!panel.panelIndex) {
-        panel.panelIndex = maxIndex++;
+        panel.panelIndex = (maxIndex++).toString();
       }
     });
   }
 
-  static getMaxPanelIndex(panels) {
-    let maxId = panels.reduce(function (id, panel) {
-      return Math.max(id, panel.panelIndex || id);
+  public static getMaxPanelIndex(panels: PanelState[]) {
+    let maxId = panels.reduce((id, panel) => {
+      return Math.max(id, Number(panel.panelIndex));
```

Comment:
For parity with the previous implementation, keep the `|| id`?

Reply:
Right on!